File: sources/academy/webscraping/scraping_basics_javascript/exercises/ikea_dimensions_regex.mjs
import * as cheerio from 'cheerio';

function parseDimensions(text) {
  const match = text.match(/(\d+)x(\d+)x(\d+) cm/);
  if (match) {
    return [match[1], match[2], match[3]];
  }
  return null;
}

const url = 'https://www.ikea.com/se/en/cat/jonaxel-system-45730/';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.plp-mastercard').toArray()) {
  const $productCard = $(element);

  const descriptionText = $productCard.find('.plp-text').text();
  const dimensions = parseDimensions(descriptionText);
  if (dimensions) {
    const price = $productCard.find('.plp-price__integer').text().replaceAll(' ', '').trim();
    console.log(`${dimensions.join(' | ')} ... ${price} SEK`);
  }
}

File: sources/academy/webscraping/scraping_basics_python/exercises/ikea_dimensions_regex.py
import re

import httpx
from bs4 import BeautifulSoup


def parse_dimensions(text):
    match = re.search(r"(\d+)x(\d+)x(\d+) cm", text)
    if match:
        return list(match.groups())
    return None


url = "https://www.ikea.com/se/en/cat/jonaxel-system-45730/"
response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for product in soup.select(".plp-mastercard"):
    description_text = product.select_one(".plp-text").text
    if dimensions := parse_dimensions(description_text):
        price = product.select_one(".plp-price__integer").text.replace(" ", "").strip()
        print(f"{dimensions[0]} | {dimensions[1]} | {dimensions[2]} ... {price} SEK")
